app/Models/Forum/Topic.php

```diff
@@ -195,6 +195,21 @@ public function nthPost($n)
         return $this->posts()->skip(intval($n) - 1)->first();
     }
 
+    public function isDoublePostBy($userId)
+    {
+        $post = $this->posts()->orderBy('post_id', 'desc')->first();
+        $postTime = $post['post_time'];
+        $now = Carbon::now();
+
+        $diffSinceLastPost = $now->diffInDays($postTime);
+
+        if (! is_null($userId) && $post['poster_id'] == $userId && $diffSinceLastPost < 3) {
+            return true;
```

Comment:
As an additional note, use property access (`$post->post_time`) instead of array access, as this is a model.